### ISSUE TITLE
Per-Example Loss Values are Wrong for BalancedCrossEntropyLoss

### ISSUE THREAD

**Issue by ramppdev** [autrainer/autrainer#84]
In our current implementation of the `BalancedCrossEntropyLoss` we calculate the class weights based on the inverse frequency of each class in the training subset, which is correct. However, we have an oversight that we normalize the weights to sum up to `1` instead of the number of classes `C`:
https://github.com/autrainer/autrainer/blob/c25132ee008508fb9cbd941724c6e9bfe86a0a4b/autrainer/criterions/criterion_wrappers.py#L28-L45

This is no issue when setting `reduction="mean"` which is the default for the [CrossEntropyLoss](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html#torch.nn.CrossEntropyLoss) implementation, as torch automatically rescales the weights in this case (can also be seen in the formulas and [here](https://github.com/pytorch/pytorch/blob/2981534f54d49fa3a9755c9b0855e7929c2527f0/aten/src/ATen/native/LossNLL.cpp#L265-L267)).

**BUT** when obtaining the per-example loss again in our `OutputsTracker`, we end up with loss values that are essentially downscaled by a factor of `C`.
This is not "severe" in a sense that the loss computation is wrong during training, but all our reported per-example loss values are definitely not correct.

**Comment by millinma** [comment]
I guess you are right. I don't seem to find that they specify this anywhere explicitly (if I didn't miss it), but I guess if we assume from the formulas that not specifying a weight results in a weight of 1, then everything should sum up to C. But I also agree that it shouldn't have impact on the training with reduction equal to "mean". For the curriculum scoring function, absolute values might be affected, but not the ranking right?

**Comment by ramppdev** [comment]
> I guess you are right. I don't seem to find that they specify this anywhere explicitly (if I didn't miss it), but I guess if we assume from the formulas that not specifying a weight results in a weight of 1, then everything should sum up to C. But I also agree that it shouldn't have impact on the training with reduction equal to "mean".

I checked the C++ implementation and they explicitly normalize to the number of classes, but you can also easily verify this by passing different weights scaled by any constant other than 0 to the loss and compare the outputs when using `reduction="mean"`.
So this had absolutely no effect on the correctness of the training results and only affected the reporting of the per-example loss value npy files.


> For the curriculum scoring function, absolute values might be affected, but not the ranking right?

This luckily has no effect on the _aucurriculum_ package (and the results) as we explicitly used a non-weighted criterion here for e.g. `CELoss`: https://github.com/autrainer/aucurriculum/blob/5c79b9c92d38da36bc8d5adea15dfba4fce1ebb0/aucurriculum-configurations/curriculum/scoring/CELoss.yaml#L1-L7
